tests/test_toolkitapi_gage.cpp

```diff
@@ -0,0 +1,104 @@
+/*
```

Comment:
@abhiramm7, what do you think of this test?

Reply:
@bemcdonnell They look awesome !